lib/src/main/java/graphql/nadel/engine/transform/partition/NadelPartitionListMerger.kt

```diff
@@ -0,0 +1,45 @@
+package graphql.nadel.engine.transform.partition
+
+import graphql.nadel.engine.transform.result.NadelResultInstruction
+import graphql.nadel.engine.transform.result.json.JsonNode
+import graphql.nadel.engine.util.unwrapNonNull
+import graphql.normalized.ExecutableNormalizedField
+import graphql.schema.GraphQLList
+import graphql.schema.GraphQLObjectType
+import graphql.schema.GraphQLOutputType
+import graphql.schema.GraphQLScalarType
+
+object NadelPartitionListMerger {
+    fun mergeDataFromList(
+        dataFromPartitionCalls: List<Any?>,
+        thisNodesData: Any?,
+        parentNodes: List<JsonNode>,
+        overallField: ExecutableNormalizedField,
+    ): List<NadelResultInstruction.Set> {
+        val parentNode = parentNodes.first()
+
+        val listDataFromPartitionCalls = dataFromPartitionCalls
+            .mapNotNull {
```

Comment:
Could be

```kotlin
.flatMap {
  if (it == null) {
    emptyList()
  } else {
    check(it is List<*>) { "Expected a list, but got ${it::class.simpleName}" }
    it
  }
}
```

Also I think that `ClassCastException` will print out the expected/actual type. So could just straight up cast it.

```kotlin
.flatMap {
  it as List<*>?
}
```